src/ui/zcl_abapgit_gui_page_diff.clas.abap

```diff
@@ -254,24 +254,23 @@ CLASS zcl_abapgit_gui_page_diff IMPLEMENTATION.
 
     DATA:
       lo_sub_filter TYPE REF TO zcl_abapgit_html_toolbar,
-      lt_types      TYPE string_table,
-      lt_users      TYPE string_table.
+      lv_user       TYPE string,
+      lt_extensions TYPE SORTED TABLE OF string WITH UNIQUE DEFAULT KEY,
+      lt_obj_types  TYPE SORTED TABLE OF string WITH UNIQUE DEFAULT KEY,
+      lt_users      TYPE SORTED TABLE OF string WITH UNIQUE DEFAULT KEY.
 
     FIELD-SYMBOLS: <ls_diff> LIKE LINE OF mt_diff_files,
                    <lv_i>    TYPE string.
-    " Get unique
+
+    " Get unique filter values
     LOOP AT mt_diff_files ASSIGNING <ls_diff>.
-      APPEND <ls_diff>-type TO lt_types.
-      APPEND <ls_diff>-changed_by TO lt_users.
+      lv_user = <ls_diff>-changed_by.
+      COLLECT <ls_diff>-type INTO lt_extensions.
+      COLLECT <ls_diff>-obj_type INTO lt_obj_types.
+      COLLECT lv_user INTO lt_users.
```

Comment:
COLLECT is complicated, and in this case it does not sum anything, perhaps use INSERT instead?

Reply:
On the contrary, COLLECT is one of the beautiful ABAP statements that other languages are missing. It's simple, fast, and produces the desired result i.e. a sorted table with unique key values. Anything else is complicated.

Reply:
INSERT into sorted table with unique key or hashed table gives the same result. Wouldn't call that complicated.

Reply:
the intention of COLLECT is to insert and "adds the values of its numeric components to the corresponding values of existing rows". If I see a COLLECT statement the first thing I check is what are the numeric components, in this case there are no numeric components, ie. the intention is not to add the values of the numeric components.

https://help.sap.com/doc/abapdocu_750_index_htm/7.50/en-US/abapcollect.htm
"If uniqueness or compression are not required, or the uniqueness is guaranteed for other reasons, the statement INSERT should be used instead."

Reply:
The first sentence in the documentation is "This statement inserts the content of a work area wa either as a single row in an internal table itab _**or**_ adds the values of its numeric components to the corresponding values of existing rows with the same primary table key". You look at only the second - admittedly complex part. I see the first.  

Also, the last sentence you mentioned says to use COLLECT, since a) uniqueness of the result _**is**_ required, b) uniqueness _**is**_ not guaranteed beforehand. 

PS: Insert into a sorted table with unique key causes CX_SY_ITAB_DUPLICATE_KEY. You will have to check with READ TABLE before if the entry already exists... which is what COLLECT does automatically.

Reply:
Hm, ordinary single INSERT into a table with unique primary key sets SY-SUBRC to 4 when a row whit the same primary key already exists. CX_SY_ITAB_DUPLICATE_KEY is only thrown when uniqueness of secondary is violated or a set INSERT violates uniqueness of primary key. Which is both not the case here therefore single INSERT is imho perfect fit for this task and no READ TABLE or catching exceptions is needed.
https://help.sap.com/doc/abapdocu_750_index_htm/7.50/en-US/abapinsert_itab.htm

Reply:
You are right about the duplicate key. I changed it to INSERT and we will just ignore subrc. Still better than what it was before. 😀 

PS: I find this "adversity for COLLECT" entertaining and wonder if other abapdevs think the same way about COLLECT.

Reply:
I've opened https://github.com/abaplint/abaplint/issues/2234 to avoid this, if the team agrees to :)

it might be a topic for https://github.com/SAP/styleguides